### DISCUSSION TITLE
Custom line numbering

### DISCUSSION THREAD

**Discussion by rojakcoder** [facebook/docusaurus#8858] [Q&A] [answered]
So we can [add line numbers to code blocks](https://docusaurus.io/docs/markdown-features/code-blocks#line-numbering) now. I'm wondering if we can change the starting line number?

So, instead of lines 1-3 for this code block:

```js
1. console.log('First line');
2. console.log('Second line');
3. console.log('Third line');
```

Can we have it starting from, say, 6:

```js
6. console.log('First line');
7. console.log('Second line');
8. console.log('Third line');
```

**Comment by slorber** [comment]
> I'm wondering if we can change the starting line number?

we don't provide this feature but you can swizzle the code block logic and implement it your self if you want

What's your motivation to do so, which API would you use to achieve it, and which docs system supports this already?

**Reply by rojakcoder** [reply]
The motivation is to indicate a specific section in the file so that readers can see which part of the file quickly. Frankly, I haven't seen this in other systems before.

**Reply by dschiller** [reply]
Absolutely agree and there is Material for MkDocs supporting that at https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-line-numbers

**Reply by slorber** [reply]
Thanks for the reference

It seems that we can achieve it quite easily, simply by adding a CSS rule such as `counter-reset: line-count 3;` anywhere above the lines.

````mdx

<div style={{"counter-reset": "line-count 3"}}>

```jsx showLineNumbers
import React from 'react';

function MyComponent(props) {
  if (props.isBar) {
    return <div>Bar</div>;
  }

  return <div>Foo</div>;
}

export default MyComponent;
```

</div>
````

Not an ideal solution though, but at least it works:

![CleanShot 2025-01-16 at 15 41 11](https://github.com/user-attachments/assets/8cc19fc7-03cf-458d-9685-8dc8d98aa417)

**Reply by slorber** [reply, marked as answer]
Implemented in https://github.com/facebook/docusaurus/pull/10846
